### DISCUSSION TITLE
[init] no_init and noncopyable

### DISCUSSION THREAD

**Discussion by huangweiwu** [wjakob/nanobind#265] [Q&A] [answered]
Does nanobind have functionalities similar to no_init and noncopyable in Boost.Python that can prevent Python from initializing and copying the bound C++ classes?

**Comment by wjakob** [comment, marked as answer]
Class bindings are `no_init` by default unless you bind some kind of a constructor. A copy constructor must also be bound explicitly.

If you completely want to avoid copies of a class being made, then that only works if the class declares a deleted copy constructor (which nanobind detects and respects).

**Reply by huangweiwu** [reply]
Get it, thanks a lot!

**Comment by huangweiwu** [comment]
I declares a deleted copy constructor for a class and use init_implicit to enable implicit conversions.
But it comes up with an error.
Attempting to reference a deleted function.
